### PR TITLE
[Bugfix] Forward seeded generator to AudioX VAE adapter

### LINKED ISSUE #4595: [Bug]: AudioX produces non-deterministic output for same-seed runs
### Your current environment

vLLM-Omni version 0.1.dev2019+g7eeef02d4
vLLM version 0.23.0

### 🐛 Describe the bug

Running 
```
python examples/offline_inference/audiox/end2end.py --tasks tv2a --video https://zeyuet.github.io/AudioX/static/samples/V2M/1XeBotOFqHA.mp4
```
generates a different audio each time, although the end2end script supplies a fixed seed. 

Here are the audio generated from different runs (2 wav files zipped; can't upload wav format here):

[wav-samples.zip](https://github.com/user-attachments/files/29182332/wav-samples.zip)

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by zzehli**
Found this issue while working on https://github.com/vllm-project/vllm-omni/issues/4540

Fixes #4595

## Repro
The newly added test fails on main branch:
```
pytest tests/e2e/offline_inference/test_audiox_model_expansion.py::test_audiox_det
erminism -v
```

Alternatively, generate two audio files with same output and verify the differences:
```
python examples/offline_inference/audiox/end2end.py --tasks tv2a --video https://zeyuet.github.io/AudioX/static/samples/V2M/1XeBotOFqHA.mp4
```

## Root cause

The seed generator isn't forwarded properly to sampler:
`AudioVaePromptAdapter.forward` called `latent_dist.sample()` with no generator, drawing noise from PyTorch's unseeded global RNG on every request.

## Fix

* Forward the user's seeded `generator` through `_encode_conditioning_tensors` →
`audio_vae_adapter` → `latent_dist.sample(generator=...)`. 
* Adds `test_audiox_determinism` regression test (fails on pre-fix code, passes after). This test generates two audio files with the same seed and test if they are the same.

`test_audiox_determinism` regression test result:
```
=========================================== warnings summary ===========================================
../.venv/lib/python3.12/site-packages/torch/jit/_script.py:365: 14 warnings
  /root/dev/.venv/lib/python3.12/site-packages/torch/jit/_script.py:365: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(

vllm_omni/version.py:55
  /root/dev/vllm-omni/vllm_omni/version.py:55: RuntimeWarning: vLLM and vLLM-Omni appear to have mismatched major/minor versions:
   --> vLLM-Omni version 0.1.dev2626+g10906e3a6
   --> vLLM version 0.26.0
  This will likely cause compatibility issues.
    warn_if_misaligned_vllm_version()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--- Running Summary
=================================== 1 passed, 15 warnings in 34.98s ====================================
```

### CODE REVIEW

**Review by hsliuustc0106 — CHANGES_REQUESTED** [review]
Pre-commit check failing. Please fix the pre-commit failures before merging.

**Comment by zzehli** [comment]
@princepride can you take a look at this bug fix? it's related to the example migrations I did.

**Comment by wtomin** [comment]
This fix feeds generator and seed to latent sampling to ensure randomness controlness control.

I think it looks good to me, but I am not sure if we should keep the accuracy test in `tests/e2e/offline_inference/test_audiox_model_expansion.py`. Hearing suggestions from @yenuo26 @congw729

**Review by wtomin — APPROVED** [review]

**Comment by Gaohan123** [comment]
Please provide local validation results. Thanks

**Comment by wtomin** [comment]
@Gaohan123 @hsliuustc0106 I think this PR is ready to be tested in CI. 

Since it introduces the deterministic test in `tests/e2e/offline_inference/test_audiox_model_expansion.py`, could you please add `diffusion-x2iat-test` label to test the nightly CI?

**Comment by zzehli** [comment]
@wtomin @Gaohan123 I updated my PR to vllm-omni 0.26. Added regression test results in the description. This PR is ready to be merged.
